kernel/main.rs

```diff
@@ -131,29 +131,23 @@ static PIT_DURATION: Duration = Duration {
 };
 
 /// Idle loop (active while idle)
-unsafe fn idle_loop() -> ! {
+unsafe fn idle_loop() {
     loop {
         asm!("cli" : : : : "intel", "volatile");
 
         let mut halt = true;
 
-        {
-            let contexts = ::env().contexts.lock();
-            for i in 1..contexts.len() {
-                if let Some(context) = contexts.get(i) {
-                    if context.interrupted {
-                        halt = false;
-                        break;
-                    }
-                }
+        for i in env().contexts.lock().iter() {
+            if i.interrupted {
+                halt = false;
+                break;
             }
         }
 
+        asm!("sti" : : : : "intel", "volatile");
+
```

Comment:
To make this not halt after an interrupt, hlt needs to be the instruction immediately after sti